test/integration/testLargeResultSet.js

```diff
@@ -106,3 +107,43 @@ describe('Large result Set Tests', function ()
     });
   });
 });
+
+describe('SNOW-743920: Large result set with ~35 chunks', function () {
+  let connection;
+  const tableName = 'test_table';
+  const sourceRowCount = 251002;
+  const createTable = `create or replace table ${tableName} (data string)`;
+  const populateData = `insert into ${tableName} select randstr(20, random()) from table (generator(rowcount =>${sourceRowCount}))`;
+  const selectAllFromOrders = `select * from ${tableName}`;
+
+  before(async () => {
+    connection = testUtil.createConnection();
+    await testUtil.connectAsync(connection);
+    await testUtil.executeCmdAsync(connection, 'alter session set ROWS_PER_RESULTSET = 1000000');
```

Comment:
This result set is larger than the number of source rows, is it correct?

Reply:
it's also connected with reported issue - but it has to be higher than number of downloaded rows